Templates/JenkinsPipeline/README.md

```diff
@@ -0,0 +1,132 @@
+  * to request the Deployment of the UrbanCode Component version via the Common Backend script [ucdDeploy.sh](../Common-Backend-Scripts/README.md#46---ucddeploysh) to the shared Development/Integration Test environment,
+  * to publish deployment log files to the Jenkins build result,
+  * to add links to the UCD Deployment request.
+ * `Workspace Cleanup` stage to clean up the workspace
+
+Depending on your selected and software analysis and deployment technology, review the definitions and (de-)/activate the appropriate steps.
+
+The pipeline is implemented as a declarative pipeline.
+
+## Prerequisites
+
+The [Common Backend scripts](../Common-Backend-Scripts/) need to be configured for the selected technologies to operate correctly.
+
+## Installation and setup of template
+
+**Note: Please work with your pipeline specialist to review the below section.**
+
+The `Jenkinsfile` can be dropped into the root folder of your Git repository and will automatically provide pipelines for the specified triggers defined in the Jenkins Multibranch Pipeline job configuration.
+
+Please review the definitions thoroughly with your Jenkins administrator. Ideally, the Jenkinsfile is converted into a [Jenkins Shared Library](https://www.jenkins.io/doc/book/pipeline/shared-libraries/) to provide stronger central controls.
+
+Step-by-step instructions to define and configure a Jenkins Multibranch Pipeline is documented at: https://www.jenkins.io/doc/book/pipeline/multibranch/
+
+
+## Pipeline usage
+
+The pipeline template implements the common clone, build, package and deploy steps to process various configurations according to the defined conventions. It leverages the a declarative checkout of the applications git repository.
+
+It is supporting the following workflows: 
+
+* automated [build pipelines for feature branches](https://ibm.github.io/z-devops-acceleration-program/docs/branching-model-supporting-pipeline#pipeline-build-of-feature-branches) with a clone and build stage,
+* the [basic pipeline](https://ibm.github.io/z-devops-acceleration-program/docs/branching-model-supporting-pipeline#the-basic-build-pipeline-for-main-epic-and-release-branches) when changes are merged into the branch `main` and
+* a [release pipeline](https://ibm.github.io/z-devops-acceleration-program/docs/branching-model-supporting-pipeline#the-release-pipeline-with-build-packaging-and-deploy-stages) to build and package the release candidate, that then can be deployed via UCD.
+
+Please review the pipeline stage `Pipeline setup` within the [template](Jenkinsfile#L165) that is configuring the subsequent stages including flags using as conditions when stages and jobs are executed. 
+
+Please make yourself familiar with the [Git branching for mainframe development](https://ibm.github.io/z-devops-acceleration-program/docs/git-branching-model-for-mainframe-dev/#characteristics-of-mainline-based-development-with-feature-branches) documentation, that this template is implementing.
```

Comment:
This template is implementing the [Git branching for mainframe development](https://ibm.github.io/z-devops-acceleration-program/docs/git-branching-model-for-mainframe-dev/#characteristics-of-mainline-based-development-with-feature-branches) model. If you haven't done yet, we would recommend to get familiar with the concepts described in this documentation.